plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/OAuth2SecurityParams.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.rest;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.rest.auth.OAuth2Properties;
+
+import javax.inject.Inject;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class OAuth2SecurityParams
+        implements SecurityParams
+{
+    private final OAuth2SecurityConfig securityConfig;
+
+    @Inject
+    public OAuth2SecurityParams(OAuth2SecurityConfig securityConfig)
+    {
+        this.securityConfig = securityConfig;
```

Comment:
Couple nits here. Let's do the checkArgument on security config once here in the constructor. Also needs a requireNonNull. 

Another small thing, config objects shouldn't be stored after construction, their fields should be. 

```suggestion
        requireNonNull(securityConfig, "securityConfig is null");
        checkArgument(
                securityConfig.getCredential().isPresent() ^ securityConfig.getToken().isPresent(),
                "OAuth2 requires either a credential or token");
        this.credential = securityConfig.getCredential();
        this.token = securityConfig.getBearer();
```